src/hasher512.cpp

```diff
@@ -203,15 +122,7 @@ namespace libtorrent
 #elif TORRENT_USE_COMMONCRYPTO
 		CC_SHA512_Init(&m_context);
 #elif TORRENT_USE_CRYPTOAPI_SHA_512
-		CryptDestroyHash(m_context);
-		if (CryptCreateHash(get_crypt_provider(), CALG_SHA_512, 0, 0, &m_context) == false)
-		{
-#ifndef BOOST_NO_EXCEPTIONS
-			throw system_error(error_code(GetLastError(), system_category()));
-#else
-			std::terminate();
-#endif
-		}
+		m_context = decltype(m_context)();
```

Comment:
this looks suspicious. this will create a hash context, then duplicate it and then tear down the first one. maybe there should just be a ``reset()`` member function on m_context